apps/src/dance/ai/cdoFieldDanceAi.js

```diff
@@ -64,6 +69,20 @@ class CdoFieldDanceAi extends GoogleBlockly.Field {
 
   showEditor_() {
     getStore().dispatch(setCurrentAiModalField(this));
+
+    // At time of writing, the modal doesn't open if clicked within a categorized toolbox.
+    // The logic to get the blocks in a categorized toolbox is different than that of an
+    // uncategorized one, so additional logic would need to be added here if that changes.
+    if (getToolboxType() === ToolboxType.UNCATEGORIZED) {
+      const isInToolbox = Blockly.getMainWorkspace()
+        .getFlyout(true)
```

Comment:
Just a terminology thing I learned doing this -- if our main workspace has an "uncategorized toolbox", `Blockly.getMainWorkspace().getToolbox()` returns null. It instead a "flyout" (via `getFlyout(true)`, which has its own workspace (with blocks, etc.)

For a workspace with an "categorized toolbox", `Blockly.getMainWorkspace().getToolbox()` returns a Toolbox. The Toolbox has an associated Flyout, which actually contains the workspace/blocks.

Reply:
Just to confirm, is an "uncategorized" toolbox the kind with lots of blocks always visible, while a "categorized toolbox" is where you see the categories and have to open one to see the blocks inside of ti?

Reply:
Correct!

Reply:
This helper got me started trying to understand this, I couldn't find super clear Blockly documentation on the topic (or I'm dense and couldn't understand it):

https://github.com/code-dot-org/code-dot-org/blob/e744327405af2e3ad8c978add9afdef42b2277ae/apps/src/blockly/addons/cdoUtils.js#L181-L195

Reply:
Just confirmed that the final level in our progression will use an "uncategorized" toolbox, so we shouldn't need to worry about the "categorized" case for now.

Reply:
Sounds good -- and to clarify, I did test adding the AI block to a "categorized toolbox", and its just that the modal doesn't open at all in that case (with or without this change). So it'd still work if we used this block on a categorized toolbox level (and the bug this PR fixes isn't relevant there, at least for now).

Reply:
Ah, got it.  Thanks for the thorough investigation, and also the clarification of all the terminology.

Reply:
+1, this is very helpful!